src/inventory_ui.h

```diff
@@ -0,0 +1,341 @@
+#ifndef INVENTORY_UI_H
+#define INVENTORY_UI_H
+
+#include <memory>
+
+#include "cursesdef.h"
+#include "input.h"
+#include "item_location.h"
+#include "item.h"
```

Comment:
It would be a really good idea to avoid, if it is possible, including `item.h` and `item_location.h`. Replace them with forward declarations.

In general, try to avoid adding header inclusions to headers as much as possible. This can slow down compilation heavily, results in implicit dependencies that are a bitch to untangle and is less explicit than forward declaring the exact classes and enums you'll be using.